lrpar/src/lib/ctbuilder.rs

```diff
@@ -862,160 +873,136 @@ where
                     #visibility
                 }
             }
-        };
-        cache_module.to_string()
+        }
     }
 
     /// Generate the main parse() function for the output file.
     fn gen_parse_function(
         &self,
         grm: &YaccGrammar<StorageT>,
         stable: &StateTable<StorageT>,
-    ) -> Result<String, Box<dyn Error>> {
-        let mut outs = String::new();
-
+    ) -> Result<TokenStream, Box<dyn Error>> {
+        let mut consts = String::new();
         // bincode format is serialized into constants which the generated
         // source code.
-        serialize_bin_output(grm, GRM_CONST_NAME, &mut outs)?;
-        serialize_bin_output(stable, STABLE_CONST_NAME, &mut outs)?;
+        serialize_bin_output(grm, GRM_CONST_NAME, &mut consts)?;
+        serialize_bin_output(stable, STABLE_CONST_NAME, &mut consts)?;
```

Comment:
So, here is one case worth keeping this in draft, because we're kind of jumping through circus hoops.
Generating these consts to source, then parsing that source to tokens, and then running the result through quote.

I can't imagine it is good given that these can be decently sized.

Reply:
Is this just a case of changing what `serialize_bin_output` does? Ultimately it's just creating a big array/vec, so should be fairly easy to change if performance is a problem.

Reply:
I noticed that upgrading bincode had these newer functions `encode_to_vec` which seemed like exactly
what we want.  Note that I kept the serde dependency, but there is also a `bincode::Encode`/`bincode::Decode` derives, might be a win but for another day.

This did however seem to increase the minimum rust compiler to 1.85.0 which is pretty new.

[3f0a91b](https://github.com/softdevteam/grmtools/pull/510/commits/3f0a91ba5bed9f3cda239697c1a26b186382a20a)

Reply:
At first I wasn't a big fan of having a dependency on 1.85.0, but there again, we are implicitly working towards a breaking release, at which point I think it's tolerable for us to have such a restriction.